Client/Frontend/Settings/SyncContentSettingsViewController.swift

```diff
@@ -147,7 +147,12 @@ class SyncContentSettingsViewController: SettingsTableViewController, FeatureFla
 
     func engineSettingChanged(_ engineName: String) -> (Bool) -> Void {
         let prefName = "sync.engine.\(engineName).enabledStateChanged"
-        return { enabled in
+        return { [unowned self] enabled in
+            // Credit card sync telemetry
+            if engineName == "creditcards" {
```

Comment:
nit: There is probably a better way of handling this using enums instead of a hard coded string.

Reply:
Good catch, I made an issue to refactor. It's not this place, but I saw it too that `RustSyncManagerAPI` class has hardcoded values for string. We can easily make it an iterable enum and then replaces `engineNames` with enum values